### PR TITLE
AudioType convert to Kotlin.

### PR DESCRIPTION
**Pull request by Lebedinsky**
## Description

Migrate to Kotlin [issue](https://github.com/mapbox/navigation-sdks/issues/166)
Add sealed AudioTypeResolver and removed separate classes.
Add tests for it

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/audio/AudioTypeResolver.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation.metrics.audio
+
+import android.content.Context
+import android.media.AudioManager
+import android.os.Build
+import com.mapbox.services.android.navigation.v5.internal.exception.NavigationException
+
+sealed class AudioTypeResolver {
```
[the remark this answers is not in the document]
Comment (Guardiola31337) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/audio/AudioTypeResolver.kt]: Should this be `internal`? And it's children?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/audio/AudioTypeResolver.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    var chain: AudioTypeResolver? = null
+
+    open fun nextChain(chain: AudioTypeResolver) {
+        this.chain = chain
+    }
+
+    abstract fun obtainAudioType(context: Context): String
+}
+
+class BluetoothAudioType : AudioTypeResolver() {
```
Comment (RingerJK) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/audio/AudioTypeResolver.kt]:
```suggestion
class Bluetooth : AudioTypeResolver() {
```
here and bottom. Sealed class is called as `AudioTypeResolver.Bluetooth(...)`

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/audio/AudioTypeChain.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation.metrics.audio
+
+class AudioTypeChain {
+
+    @JvmOverloads
```
[the remark this answers is not in the document]
Comment (Lebedinsky) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/audio/AudioTypeChain.kt]: Not yet, cause we still use it in `NavigationUtils` (it is still java), but I marked this method via `Fixme` to fix it later when we migrate `NavigationUtils` to Kotlin.

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/audio/AudioTypeResolver.kt [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+internal sealed class AudioTypeResolver {
+
+    companion object {
+        const val BLUETOOTH = "bluetooth"
+        const val HEADPHONES = "headphones"
+        const val SPEAKER = "speaker"
+        const val UNKNOWN = "unknown"
+    }
+
+    var chain: AudioTypeResolver? = null
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/audio/AudioTypeResolver.kt]: Could we make this non-nullable and `internal`? That way we will be able to remove all the `throw `NavigationException`s in the different `AudioTypeResolver`s. Also `nextChain` is called from `AudioTypeChain` which is `internal` so we can ensure that `AudioTypeResolver` won't be `null` (actually they're non-null 👀 `AudioTypeChain#setup`.

**Pull request merged**